### PR TITLE
infra: changelog generation

### LINKED ISSUE #1244: Add more changelog sections
### Clear and concise description of the problem

The changelog.md does not contain hints regarding locale updates and refactors.
Especially the later are quite important as these could be considered (future) breaking changes.

### Suggested solution

Add more sections to the changelog.

### Alternative

Remove all of the commit prefixes except:

- feat
- fix
- chore
- docs
- test
- infra

### Additional context

- The changelog.md should be updated retrospectively.

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
This PR introduces configuration for our CHANGELOG file according to the [`standard-version` package](https://github.com/conventional-changelog/standard-version#configuration).

You can see what commands would be run, without committing to git or updating files by using the dry run mode:
1. switch up project locally (if you haven't already)
2. switch to this branch
3. run `npm run release -- --dry-run`

I will provide a sample below.

Fixes #1244

### CODE REVIEW

**Comment by xDivisionByZerox** [comment]
## Current Dry Run Sample (will be updated regularly)

<details>
<summary>Changelog</summary>

## [8.0.0](https://github.com/faker-js/faker/compare/v8.0.0-alpha.0...v8.0.0) (2023-01-21)


### Features

* **company:** add realistic Hungarian company names ([#1758](https://github.com/faker-js/faker/issues/1758)) ([2e6b136](https://github.com/faker-js/faker/commit/2e6b136c61a7cfc799c72a86a38ca4a505d94e3c))
* **helpers:** add new faker.helpers.weightedArrayElement ([#1654](https://github.com/faker-js/faker/issues/1654)) ([59824e6](https://github.com/faker-js/faker/commit/59824e66ad1e14be2d61b8425e8f8e9547863565))
* **helpers:** allow empty string in fake ([#1679](https://github.com/faker-js/faker/issues/1679)) ([f0d2ffb](https://github.com/faker-js/faker/commit/f0d2ffb256e5aefb98d4f38905ef081c8e6af43e))
* **locales:** add mergeLocales utility ([#1707](https://github.com/faker-js/faker/issues/1707)) ([788fce0](https://github.com/faker-js/faker/commit/788fce048f0233552df47f3407c06d8f83c32b0c))
* **number:** add binary and octal random number generation ([#1708](https://github.com/faker-js/faker/issues/1708)) ([d3229fc](https://github.com/faker-js/faker/commit/d3229fcdf28b5a1abdbc44a7bcdde934bd472bf2))
* **person:** add short bio ([#1696](https://github.com/faker-js/faker/issues/1696)) ([2a2e054](https://github.com/faker-js/faker/commit/2a2e054a16a45bc5cd56bdd923b7f5140eb12699))
* **person:** split prefix into gendered versions ([#1665](https://github.com/faker-js/faker/issues/1665)) ([27dff93](https://github.com/faker-js/faker/commit/27dff93aa27d755874aa5022c78f17ff8e9cf7e0))  
* **string:** add binary and octal random string generation ([#1710](https://github.com/faker-js/faker/issues/1710)) ([80e0a96](https://github.com/faker-js/faker/commit/80e0a9688099677a74614ccbedc9231b6382c4eb))
* **string:** generate string from characters ([#1717](https://github.com/faker-js/faker/issues/1717)) ([47b2cfc](https://github.com/faker-js/faker/commit/47b2cfc76b790647e398bf9883368a10b2ff5a68))      
* **string:** nanoid ([#1716](https://github.com/faker-js/faker/issues/1716)) ([099e76c](https://github.com/faker-js/faker/commit/099e76ce0fb180beb5fd62d72a07c236e04cdca0))


### Bug Fixes

* **internet:** fix invalid emails in some locales ([#1746](https://github.com/faker-js/faker/issues/1746)) ([1e4e869](https://github.com/faker-js/faker/commit/1e4e8699e59f3b5b9c1e1d6ad9b89ee4cc254e95)) 
* **locale:** filter inappropriate words in the en locale ([#1745](https://github.com/faker-js/faker/issues/1745)) ([dfa647d](https://github.com/faker-js/faker/commit/dfa647dc07ea25f962c4df913bae2de082e8acc4))


### New Locales

* **locale:** add Luxembourg (French) locale ([#1693](https://github.com/faker-js/faker/issues/1693)) ([c66f7bd](https://github.com/faker-js/faker/commit/c66f7bd70f930c4a080cc24cc8d09baab260f6d0))       
* **locale:** extend english company ([#1701](https://github.com/faker-js/faker/issues/1701)) ([f0e859b](https://github.com/faker-js/faker/commit/f0e859ba3033e54aefd71bf194da574593bc1631))
</details>

File: .versionrc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "types": [
+    { "type": "feat", "scope": "locale", "section": "New Locales" },
+    { "type": "feat", "section": "Features" },
+    { "type": "refactor", "scope": "locale", "section": "Changed Locales" },
+    { "type": "fix", "section": "Bug Fixes" },
+    { "type": "chore", "hidden": true },
+    { "type": "docs", "hidden": true },
+    { "type": "refactor", "hidden": true }
```
Comment (xDivisionByZerox) [inline .versionrc:9]: I could even go as far as `{ "type": "refactor", "scope": "<moduleName>", "section": "Changes in <moduleName>" }` which could be very nice tbh.

Only point would be to update the expression if a new module is introduced (which can be documented).

I would be willing to provide all of that, but what your decision on that.

File: .versionrc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "types": [
+    { "type": "feat", "scope": "locale", "section": "New Locales" },
```
Comment (xDivisionByZerox) [inline .versionrc:3]: Should this section be called `"New Locale Features"`? Or should we allow `feat(locale)` only if we introduce a new locale?

**Comment by Shinigami92** [comment]
That looks somewhat interesting 😲

But should we put "effort" into configuring it when we actually want to move away from that dependency?
- #1096

Please be sure that the title of that linked issue is not really up to date anymore, as I'm not sure if we really want to use `changesets` or an alternative like described in the issue itself

But I'm heavily and in conflict with myself what would be best for the project in the long run as I didn't have build up enough knowledge yet

**Comment by xDivisionByZerox** [comment]
> But should we put "effort" into configuring it when we actually want to move away from that dependency?
> [...]
> But I'm heavily and in conflict with myself what would be best for the project in the long run as I didn't have build up enough knowledge yet

This didn't took too long to research. So "putting effort" into it would only be if we accept [my suggestion](https://github.com/faker-js/faker/pull/1304#discussion_r956458876). 

Currently, this PR fixes our problem with locale changes not showing up. This should have priority.
So if we change our release package, it should also be able to be customizable as  well.

File: .versionrc (hunk 2 above)
Comment (Shinigami92) [reply .versionrc:3]: More like locale changes :thinking: 
Sometimes not new but just e.g. typo fixes or even removals are also affecting the locales

**Comment by Shinigami92** [comment]
> > But should we put "effort" into configuring it when we actually want to move away from that dependency?
> > [...]
> > But I'm heavily and in conflict with myself what would be best for the project in the long run as I didn't have build up enough knowledge yet
> 
> This didn't took too long to research. So "putting effort" into it would only be if we accept [my suggestion](https://github.com/faker-js/faker/pull/1304#discussion_r956458876).
> 
> Currently, this PR fixes our problem with locale changes not showing up. This should have priority. So if we change our release package, it should also be able to be customizable as well.

Yes, but on the other side we documented and changed the way how we name PR titles and so this fixed the problems with locales as well + it provided us a really helpful convention/structure of PR titles
So there is not so much randomness in the titles anymore
But this PR is not only good for showing up locales, but actually separating them from the features-segment, and that could be a good thing :thinking:

File: .versionrc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (Shinigami92) [inline .versionrc:1]: Not yet decided if we want to go with this PR, but if so, I would prefer using `.versionrc.json` so editors can easily pick up the file language by the extension of the file

**Comment by xDivisionByZerox** [comment]
We wait with merging till v8, to start with a "fresh" new  changelog style.

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**